EIPS/eip-crosschain-id.md

```diff
@@ -0,0 +1,131 @@
+challenges for inter-chain interoperability, particularly when these chains are
+heterogeneous and incompatible.
+Enterprise Ethereum Alliance formed Crosschain Interoperability Task Force (CITF) to look into common crosschain problems and solutions. CITF team noticed that there is a lack of unique identifier to charaterise and describe a blockchain. Several proprosals were discussed in EEA Crosschain Interoperability Task Force meetings and discussions.  We have considered various alternative specifications such as using a random unique hex string to represent a blockchain.  The drawback of this method is that the random id can not be used to verify a blockchain's intrinsic identity such as the blockhash of the genesis block.  A second alternative is simply using a genesis blockhash to represent a blockchain id for crosschain operations.  The drawback of this is that this id does not have information about the property of the blockchain and it has problem when a blockchain is forked into two blockchain.  After various discussion, we finally came up with hybrid crosschain identity specification that allows a blockchain to be identified and verified both manually and through computer programs.
+
+## Backwards Compatibility
+
+Crosschainid can be backward compatible with EIP-155.  The crosschain id contains a 4 byte segment to record the chainid based on EIP-155.
+
+## Test Cases
+
+The test cases should cover the generation of crosschain id, the parsing of an crosschain id to ensure the various segments of the crosschain id meets the specification.
+For the crosschain registration smart contract, the test cases should cover the testing of all functions and also need to ensure that the security of each functions are properly implemented.
+
+## Implementation
+
+The implementation requires the following
+
+* Generate the crosschain id based on the format defined in the specification.  
+* Parse the crosschain id based on the format defined in the spefication.
+* For metadata asssociated with the crosschainid, the medadata needs to be registered with a smart contract.  The smart contract should implement the functions defined in the speficification.
+* Once a smart contract is deployed, the owner of the smart contract can publish the smart contract address, ABI and also crosschain id specification version.
+* Once a crosschain id is registered in the smart contract with extended metadata, there should be a waiting period for the community to review the registration and raise objection if there is inconsistency.
+
+## Security Considerations
```

Comment:
There are few formatting ideas in this section, it would be best to choose one format and stick with it. For example: 

```
problem: solution
or
#### problem
solution
```

Reply:
I recommend `###` rather than `####` to keep ToC tree neat.  😄

Reply:
Seems the format has been changed.